From 65bc12299239c36fe85d093b992c36f98732ff44 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Fri, 24 Jun 2011 23:44:26 +0800
Subject: [PATCH 226/505] ENGR00151160 vpu: Fix image display incorrect with
 vpu dering

vpu decoder: Fix output image display incorrect when vpu dering
is enabled.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/dec.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 7ef15b9..ea2d14d 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -717,8 +717,12 @@ decoder_start(struct decode *dec)
 		 * We convert it to clockwise, which is consistent with V4L2
 		 * rotation angle strategy.
 		 */
-		if (rot_angle == 90 || rot_angle == 270)
-			rot_angle = 360 - rot_angle;
+		if (rot_en) {
+			if (rot_angle == 90 || rot_angle == 270)
+				rot_angle = 360 - rot_angle;
+		} else
+			rot_angle = 0;
+
 		vpu_DecGiveCommand(handle, SET_ROTATION_ANGLE,
 					&rot_angle);
 
@@ -726,10 +730,10 @@ decoder_start(struct decode *dec)
 		vpu_DecGiveCommand(handle, SET_MIRROR_DIRECTION,
 					&mirror);
 
-		if (rot_en || dering_en) {
+		if (rot_en)
 			rot_stride = (rot_angle == 90 || rot_angle == 270) ?
 					fheight : fwidth;
-		} else
+		else
 			rot_stride = fwidth;
 		vpu_DecGiveCommand(handle, SET_ROTATOR_STRIDE, &rot_stride);
 	}
-- 
1.8.0

